packages/fast-components-react-base/src/slider/slider.tsx

```diff
@@ -0,0 +1,1078 @@
+            activeThumb: null,
+            isDragging: false,
+            isIncrementing: false,
+        };
+    }
+
+    /**
+     * React lifecycle methods
+     */
+    public componentWillUnmount(): void {
+        this.suspendActiveOperations();
+    }
+
+    public componentDidUpdate(prevProps: SliderProps): void {
+        if (prevProps.disabled !== this.props.disabled && this.props.disabled) {
+            this.suspendActiveOperations();
+        }
+
+        if (
+            prevProps.value !== this.props.value ||
+            prevProps.constrainedRange !== this.props.constrainedRange ||
+            prevProps.step !== this.props.step
+        ) {
+            this.suspendActiveOperations();
+            this.updateValues(this.state.lowerValue, this.state.upperValue);
+        }
+
+        if (prevProps.mode !== this.props.mode) {
+            this.suspendActiveOperations();
+
+            switch (this.props.mode) {
+                case SliderMode.adjustBoth:
+                    break;
+
+                case SliderMode.adustLowerValue:
+                    this.updateValues(this.state.lowerValue, this.props.range.maxValue);
+                    break;
+
+                case SliderMode.adustUpperValue:
+                    this.updateValues(this.props.range.minValue, this.state.upperValue);
+                    break;
+
+                case SliderMode.singleValue:
+                    this.updateValues(this.state.upperValue, this.state.upperValue);
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        this.updateDirection();
+        return (
+            <div
+                {...this.unhandledProps()}
+                ref={this.rootElement}
+                className={this.generateClassNames()}
+            >
+                <SliderContext.Provider
+                    value={{
+                        sliderOrientation: this.props.orientation,
+                        sliderMode: this.props.mode,
+                        sliderUpperValue: this.state.upperValue,
+                        sliderLowerValue: this.state.lowerValue,
+                        sliderConstrainedRange: this.props.constrainedRange,
+                        sliderActiveThumb: this.state.activeThumb,
+                        sliderIsDragging: this.state.isDragging,
+                        sliderIsIncrementing: this.state.isIncrementing,
+                        sliderValueAsPercent: this.valueAsPercent,
+                        sliderDirection: this.direction,
+                    }}
+                >
+                    <div
+                        className={get(
+                            this.props.managedClasses,
+                            "slider_layoutPanel",
+                            ""
+                        )}
+                        style={{
+                            position: "relative",
```

Comment:
can you just add this to the JSS?

Reply:
I hardcoded these values because they are IMO intrinsic to how the component functions and I don't think other configurations would even work.